jupyterhub/assignments/intake-dashboard/whats_happening_where.ipynb

```diff
@@ -0,0 +1,130 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "\n",
+    "warnings.filterwarnings(\"ignore\", category=FutureWarning)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext sql\n",
+    "%config SqlMagic.feedback = False\n",
+    "%config SqlMagic.displaylimit = 0\n",
+    "%config SqlMagic.displaycon = False"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import datetime\n",
+    "\n",
+    "start = datetime.datetime.now() - datetime.timedelta(days=90)\n",
+    "end = datetime.datetime.now()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%sql report_locations <<\n",
+    "\n",
+    "SELECT location_name, COUNT(*) count\n",
+    "FROM cts_forms_report\n",
+    "WHERE location_name notnull\n",
+    "GROUP BY location_name\n",
+    "HAVING COUNT(*) > 1\n",
+    "ORDER BY count desc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_location_count(location):\n",
+    "    reports = %sql SELECT * \\\n",
+    "    FROM report_districts \\\n",
+    "    WHERE location_name = '{{location}}';\n",
+    "    \n",
+    "    return reports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.display import display, HTML\n",
+    "\n",
+    "readable_start = start.strftime(\"%B %d, %Y\")\n",
+    "readable_end = end.strftime(\"%B %d, %Y\")\n",
+    "reports = report_locations.DataFrame()\n",
+    "locations = []\n",
+    "for i, location in enumerate(reports[reports['location_name'].notnull()]['location_name'].values):\n",
+    "    count = reports[reports['location_name'] == location]['count'].values[0]\n",
+    "    if i < 10:\n",
```

Comment:
you can remove this by putting a `LIMIT` on the query above.

It's always better to do that kind of thing in the SQL, because you're technically pulling the full list of locations into memory with this query